### ISSUE TITLE
Can't get Tegola to work in Kuberentes

### ISSUE THREAD

**Issue by thesteve0** [go-spatial/tegola#647]
I am trying to get the tegola container working in Kuberenets 1.13 (OpenShift 1.13)

I have made a configap with the config.toml and I bind it to a file mapping at /opt/tegola_config. I checked in the running container and it is appears in the right location in the pod

```
[spousty@localhost ~]$ oc debug  tegola-5-h5qbl
Defaulting container name to tegola.
Use 'oc describe pod/tegola-5-h5qbl-debug -n demo' to see all of the containers in this pod.

Debugging with pod/tegola-5-h5qbl-debug, original command: /opt/tegola serve
Waiting for pod to start ...
If you don't see a command prompt, try pressing enter.

/opt $ env | grep config
config=/opt/tegola_config/config.toml
/opt $ cat /opt/tegola_config/config.toml
[webserver]
port = ":9090"              # port to bind the web server to. defaults ":8080"

   [webserver.headers]
   Access-Control-Allow-Origin = "*"
   Cache-Control = "no-cache, no-store, must-revalidate"

[cache]                     # configure a tile cache
type = "file"               # a file cache will cache to the local file system
basepath = "/tmp/"          # where to write the file ...
```

I also pass in the path as an environment variable, but tegola is still not seeing my config.toml. 

Here is the spec
spec:
  containers:
    - args:
        - serve
      command:
        - /opt/tegola

But here is the error I am receiving when it tries to run

```
Error: config file at location (config.toml) not found!
Usage:
  tegola serve [flags]

Aliases:
  serve, server

Flags:
  -h, --help          help for serve
  -n, --no-cache      turn off the cache
  -p, --port string   port to bind tile server to (default ":8080")

Global Flags:
      --config string   path to config file (default "config.toml")

2019-09-28 00:34:38 [INFO] root.go:61: Loading config file: config.toml
2019-09-28 00:34:38 [INFO] config.go:207: loading local config (config.toml)
config file at location (config.toml) not found!
```

**Comment by gdey** [comment]
Quickly scanning seems this might be related to: https://github.com/go-spatial/tegola/issues/644#issuecomment-529189458

@thesteve0 could you make sure the config is correct. The instructions for docker is wrong

**Comment by ARolek** [comment]
@thesteve0 I'm not well versed on Kubernetes, but can you add command flags to the `args` block so it's issuing the command `/opt/tegola serve --config /opt/tegola_config/config.toml`? It looks like you're setting an env var for the config but tegola is expecting a command flag if the config is not in the default location (config.toml next to the binary)

**Comment by thesteve0** [comment]
So I tried two different techniques. I tried mapping the file to  /opt/ but
that didn't work because it overwrote the binary.
So then I worked on the config and I couldn't get it to work no matter how
I tried quoting arg it keeps saying

```
unknown flag: --config /opt/tegola_config/config.toml
Error: unknown flag: --config /opt/tegola_config/config.toml
Usage:
tegola serve [flags]

Aliases:
serve, server

Flags:
-h, --help help for serve
-n, --no-cache turn off the cache
-p, --port string port to bind tile server to (default ":8080")

Global Flags:
--config string path to config file (default "config.toml")
```

**Comment by ARolek** [comment]
@thesteve0 hmmm, If you want we can probably move through this quick via chat. Want to [join the Go Spatial Slack](https://join.slack.com/t/go-spatial/shared_invite/enQtNjY4ODcwNjc0MzU3LTJmNDhmNWYwNjUzMDkzNmU5M2UyNzVjY2Q2NWEyODAwYTFlYjUzNWZkY2Q0NDIzZjdkYmU1OTk3OGU0MGUzNWE) and see if we can figure this out?

**Comment by ingenieroariel** [comment]
I have seen this before. Looks related to ENTRYPOINT vs CMD options. Play
with creating a local Docker file overriding both.

-a

**Comment by thesteve0** [comment]
Yeah that is what I am thinking as well.
It would be nice though to have a go-spatial one that was actually built
for kube use. I could help out with that one if you want.

**Comment by ARolek** [comment]
@stvno yes, please! I have not ventured into the world of Kube too much. I'm happy to support anyway I can. I think it would be great to have a Dockerfile that would make running tegola easier with Kube as well as associated docs. LMK how I can help.

**Comment by thesteve0** [comment]
Ok traveling for work now (with just my windows machine). I will try to get
to this next week.

**Comment by ARolek** [comment]
@thesteve0 no worries! let's pick this up when you return. Safe travels.

**Comment by gdey** [comment]
@thesteve0 did you ever get back to this, it would be great to have a Dockerfile that works with Kube.

**Comment by timwaters** [comment]
if it helps, within a deployment I use:

```
 command: ["/bin/sh"]
 args: ["-c", "/opt/tegola serve --config /opt/tegola_config/config.toml"]
```
